File: app/Models/Transaction.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Transaction extends Model
{
    use HasFactory;

    protected $fillable = ['order_id', 'total_harga', 'metode_pembayaran', 'status'];

    /**
     * Relasi ke Order
     */
    public function order()
    {
        return $this->belongsTo(Order::class);
    }
}

File: go-backend/model/transaction.go
package model

import "time"

type Transaction struct {
	ID         int64     `json:"id"`
	OrderID    int64     `json:"order_id"`
	TotalHarga float64   `json:"total_harga"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`
}
